src/tools/aml-language-server/messageTypes.ts

```diff
@@ -0,0 +1,250 @@
+import * as vscode from 'vscode-languageserver';
+
+export interface InitializeParams extends vscode.InitializeParams {
+    capabilities: ClientCapabilities;
+}
+
+/**
+ * Settings to enable plugin loading
+ */
+export interface PluginSettings {
+    allowLocalPluginLoads: boolean;
+    globalPlugins: string[];
+    pluginProbeLocations: string[];
+}
+
+export interface ClientCapabilities extends vscode.ClientCapabilities {
+    /**
+     * The client provides support for workspace/xfiles.
+     */
+    xfilesProvider?: boolean;
+
+    /**
+     * The client provides support for textDocument/xcontent.
+     */
+    xcontentProvider?: boolean;
+
+    /**
+     * The client provides support for cache/get and cache/set methods
+     */
+    xcacheProvider?: boolean;
+
+    /**
+     * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+     */
+    streaming?: boolean;
+}
+
+export interface ServerCapabilities extends vscode.ServerCapabilities {
+    xworkspaceReferencesProvider?: boolean;
+    xdefinitionProvider?: boolean;
+    xdependenciesProvider?: boolean;
+    xpackagesProvider?: boolean;
+
+    /**
+     * The server supports receiving results solely through $/partialResult notifications for requests from the server to the client.
+     */
+    streaming?: boolean;
+}
+
+export interface InitializeResult extends vscode.InitializeResult {
+    capabilities: ServerCapabilities;
+}
+
+export interface TextDocumentContentParams {
+    /**
+     * The text document to receive the content for.
+     */
+    textDocument: vscode.TextDocumentIdentifier;
+}
+
+export interface WorkspaceFilesParams {
+    /**
+     * The URI of a directory to search.
+     * Can be relative to the rootPath.
+     * If not given, defaults to rootPath.
+     */
+    base?: string;
+}
+
+/**
+ * Represents information about a programming construct that can be used to identify and locate the
+ * construct's symbol. The identification does not have to be unique, but it should be as unique as
+ * possible. It is up to the language server to define the schema of this object.
+ *
+ * In contrast to `SymbolInformation`, `SymbolDescriptor` includes more concrete, language-specific,
+ * metadata about the symbol.
+ */
+export interface SymbolDescriptor {
```

Comment:
Did you write this file yourself or is it a copy paste?

Reply:
This is a set of types that were not included in the vscode npm packages, copied from the LSP standard (here: https://microsoft.github.io/language-server-protocol/specification ).

I believe it is under creative commons ( https://github.com/microsoft/language-server-protocol/blob/gh-pages/License.txt ) and so does not need referencing?

Reply:
It's worth including a link to the source, so other readers of this code know where to find the original (and to help update it if/when we need to).